code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/RegionMonitor.java

```diff
@@ -90,16 +91,19 @@ public void run() {
                     continue;
                 }
                 // environment not present or its uuid changed 
-                ExternalProject externalProject = null;
-                String externalProjectKey = String.format("%s$%s", targetRegion, link.getLinkedAccount());
+                ExternalProjectResponse externalProjectResponse = null;
+                String externalProjectKey = String.format("%s:%s", targetRegion.getName(), link.getLinkedAccount());
                 if(externalProjectMap.containsKey(externalProjectKey)) {
-                    externalProject = externalProjectMap.get(externalProjectKey);
+                    externalProjectResponse = externalProjectMap.get(externalProjectKey);
                 } else {
-                    externalProject = RegionUtil.getTargetProjectByName(targetRegion, link.getLinkedAccount(), jsonMapper);
-                    externalProjectMap.put(externalProjectKey, externalProject);
+                    externalProjectResponse = RegionUtil.getTargetProjectByName(targetRegion, link.getLinkedAccount(), jsonMapper);
+                    externalProjectMap.put(externalProjectKey, externalProjectResponse);
                 }
-                String storedUUID = DataAccessor.fieldString(link, "uuid");
-                if(externalProject == null || (storedUUID!=null && !externalProject.getUuid().equals(storedUUID))) {
+
+                String storedUUID = DataAccessor.fieldString(link, "linked_account_uuid");
+                ExternalProject externalProject = externalProjectResponse.getExternalProject();
+                if((externalProject == null && externalProjectResponse.getStatusCode()==200)
```

Comment:
Nit: can we assign 2 conditions to vars, and then use the vars in if clause? So the condition one would be `boolean notfound` and condition 2 `uuidsDoNotMatch` or something